www/src/components/account/billing/ConfirmPayment.tsx

```diff
@@ -0,0 +1,276 @@
+              const paymentMethodId
+                = typeof setupIntent?.payment_method === 'string'
+                  ? setupIntent?.payment_method
+                  : setupIntent.payment_method?.id
+
+              if (paymentMethodId) {
+                makeDefaultMutation({ variables: { id: paymentMethodId } })
+                  .then(() => {
+                    if (cancelled) {
+                      return
+                    }
+                    setDefaultPaymentId(paymentMethodId)
+                  })
+                  .catch(e => {
+                    if (cancelled) {
+                      return
+                    }
+                    setError(e)
+                  })
+              }
+              else {
+                setError(new Error('No payment method id returend'))
+              }
+            }
+          })
+          .catch(e => {
+            if (cancelled) {
+              return
+            }
+            setError(e)
+          })
+      }
+      catch (e) {
+        setError(e as Error)
+      }
+
+      return () => {
+        cancelled = true
+      }
+    }
+  }, [clientSecret, makeDefaultMutation, stripe])
+
+  return { error, setupIntent }
+}
+
+export function UpgradeSuccessMessage() {
+  return (
+    <P>
+      Welcome to the Plural Professional plan! You now have access to groups,
+      roles, service accounts, and more.
+    </P>
+  )
+}
+
+export default function ConfirmPayment() {
+  const [searchParams] = useSearchParams()
+  const location = useLocation()
+  const navigate = useNavigate()
+
+  const clientSecret = searchParams.get('setup_intent_client_secret')
+  const plan: PlanType
+    = searchParams.get('plan') === 'yearly' ? 'yearly' : 'monthly'
+  const { proPlatformPlan, proYearlyPlatformPlan }
+    = useContext(PlatformPlansContext)
+  const planId
+    = plan === 'yearly' ? proYearlyPlatformPlan.id : proPlatformPlan.id
+
+  const [upgradeSuccess, setUpgradeSuccess] = useState(false)
+  const [isOpen, setIsOpen] = useState(!!clientSecret)
+
+  const onClose = useCallback(() => {
+    setIsOpen(false)
+    navigate(location.pathname)
+  }, [location.pathname, navigate])
+
+  useEffect(() => {
+    if (upgradeSuccess && !isEmpty(searchParams.values)) {
+      navigate(location.pathname)
+    }
+  }, [location.pathname, navigate, searchParams.values, upgradeSuccess])
+
+  // Upgrade mutation
+  const [upgradeMutation, { error: upgradeError }]
+    = useUpgradeToProfessionalPlanMutation({
+      variables: { planId },
+      refetchQueries: [namedOperations.Query.Subscription],
+      onCompleted: () => {
+        setUpgradeSuccess(true)
+      },
+    })
+
+  // Confirm payment
+  const onComplete = useCallback(() => {
+    upgradeMutation()
+  },
+  [upgradeMutation])
+  const { error: setDefaultError, setupIntent } = useConfirmAndSetDefault({
+    clientSecret,
+    onComplete,
+  })
+
+  const error = upgradeError || setDefaultError
+
+  useEffect(() => {
```

Comment:
I'm pretty sure this should no longer be an issue if you want to drop that annoying error handling

Reply:
Are you talking about the 'account_id has already been taken' error? It can still crop up if the user refreshes after the upgrade has already succeeded. Though I guess I could just check if they're already upgraded before attempting the mutation.

Reply:
yea i think it should be fine, on the refresh, it should load the page w/ the subscription pinned to the pro plan so i'd assume it wouldn't cause an issue right?